frontend/app/src/types/filtering.ts

```diff
@@ -37,12 +38,21 @@ export type MatchedKeyword<T extends string> = {
   [key in T]?: string | string[];
 };
 
-export interface Suggestion {
-  readonly index: number;
-  readonly total: number;
-  readonly key: string;
-  readonly asset: boolean;
-  readonly value: AssetInfoWithId | string;
+export const Suggestion = z.object({
+  index: z.number(),
+  total: z.number(),
+  key: z.string(),
+  asset: z.boolean(),
+  value: AssetInfoWithId.or(z.string())
+});
+
+export type Suggestion = z.infer<typeof Suggestion>;
+
+export enum SavedFilterLocation {
+  HISTORY_TRADES = 'HISTORY-TRADES',
+  HISTORY_DEPOSITS_WITHDRAWALS = 'HISTORY-DEPOSITS-WITHDRAWALS',
+  HISTORY_TRANSACTIONS = 'HISTORY-TRANSACTIONS',
+  HISTORY_LEDGER_ACTIONS = 'HISTORY-LEDGER-ACTIONS'
```

Comment:
Are these values in the enum going to be serialized to JSON?

Reply:
hmm wdym serialized to JSON?

Reply:
![image](https://user-images.githubusercontent.com/2269732/220587648-2621aa13-b2c0-40d9-acea-8b37e6c1ec7e.png)

Reply:
Btw for the route history only the `key` and `value` is stored, why do we need to serialize everything here?

Reply:
I need to think a bit about it but generally, our approach here should be to store the least possible amount of information that can help us restore the filter.

Reply:
okay, let me see how I can reduce the information stored there

Reply:
> ![image](https://user-images.githubusercontent.com/2269732/220587648-2621aa13-b2c0-40d9-acea-8b37e6c1ec7e.png)

I thought I could get away from the conversion by using kebab case

Reply:
> 

Yeah I think we can store it like that, the ones with multiple usages will be stored as array instead single string (for the value)